test/contracts/stdlib_tests.aes

```diff
@@ -0,0 +1,73 @@
+// Collection of non-exhaustive tests. Feel free to add stuff here.
+
+include "Func.aes"
+include "Option.aes"
+include "List.aes"
+include "Pair.aes"
+include "Triple.aes"
+
+contract StdTest =
+  function assert(p) = switch(p)
+    (t, msg) => if(t) () else abort(msg)
+
+
+  // dear programmer from the future – this was written when `==` operator
+  // was not working on any ADT. If you are capable of doing it better – do it.
+  function list_int_cmp(l1 : list(int), l2 : list(int)) = switch((l1, l2))
+    ([], []) => true
+    (h1::t1, h2::t2) => h1 == h2 && list_int_cmp(t1, t2)
+    _ => false
+
+  function option_int_cmp(a : option(int), b) = switch((a, b))
+    (None, None) => true
+    (Some(x), Some(y)) => x == y
+    _ => false
+
+  function list_int_string_cmp(l1 : list(int * string), l2 : list(int * string)) = switch((l1, l2))
+    ([], []) => true
+    ((i1, s1)::t1, (i2, s2)::t2) => s1 == s2 && i1 == i2 && list_int_string_cmp(t1, t2)
+    _ => false
+
+
+  entrypoint test() = List.foreach(assert,
+    [ (Func.id(1) == 1, "id")
+    , (Func.const(1)(0) == 1, "const")
+    , (Func.uncurry2(Func.const)(1,0) == 1, "uncurry2")
+    , (Func.flip(Func.uncurry2(Func.const))(0,1) == 1, "flip")
+    , (Func.curry2((x, y) => x + y)(0)(1) == 1, "curry")
+    , (Func.recur((r, n) => if (n < 2) 1 else n * r(n-1))(5) == 120, "recursor")
+    , (Func.iter(5, (x) => x + 1)(0) == 5, "iter")
+    , (Func.tuplify2((a, b) => a + b)((2, 3)) == 5, "tuplify")
+
+    , (List.is_empty([]), "empty")
+    , (option_int_cmp(List.first([1,2,3,4]), Some(1)), "list first")
+    , (option_int_cmp(List.find((x) => x > 0, [-1, -2, 3, -1]), Some(3)), "find")
+    , (list_int_cmp(List.replace_at(2, 0, [1,1,1,1,1]), [1,1,0,1,1]), "replace at")
+    // following test seems to break the FATE runtime. It definitely SHOULD work.
+    // , (list_int_cmp(List.foldr((e, cont) => (x) => cont(e::x), (x) => x, [1,2,3])([]), [3,2,1]), "foldr")
+    , (List.foldl((a, x) => a - x, 0, [1,2,3]) == -6, "foldl")
+    , (list_int_cmp(List.reverse([1,2,3]), [3,2,1]), "erslist_int_cmp(e")
```

Comment:
typo?

Reply:
Yep, it looks like I overused emacs macros